### DISCUSSION TITLE
Emissão de MDFe: ordem dos campos de <veicReboque> influencia na autorização do documento

### DISCUSSION THREAD

**Discussion by edivaldoramos** [wmixvideo/nfe#872] [Geral]
Olá pessoal.
Tava fazendo uns testes de geração de documento mdfe em ambiente de producão e notei um comportamento bastante curioso.
A lib monta os dados de Veiculo Reboque na seguinte ordem:
```
<veicReboque>
    <placa>QKL8049</placa>
    <tara>23000</tara>
    <tpCar>02</tpCar>
    <UF>TO</UF>
    <capKG>23000</capKG>
</veicReboque> 
```
entretanto, o documento não é autorizado e retorna a mensagem **580-Rejeição: Falha no Schema XML específico para o modal** .
 
Dessa forma, considerei ordenar os campos colando UF no final e a sefaz autorizou o documento.
```
<veicReboque>
    <placa>QKL8049</placa>
    <tara>23000</tara>
    <capKG>23000</capKG>
    <tpCar>02</tpCar>
    <UF>TO</UF>
</veicReboque>
```
Alguem já conhece essa situação e caso sim, teríamos uma forma de contornar esse problema ?

**Comment by aguinaldofryder** [comment]
Olá Edivaldo, 
Realmente a ordem dos campos pode causar rejeição. 
Já tive esse problema em outros campos da NF-e.
Para resolver basta trocar a ordem dos atriutos e submeter um PR, como eu fiz no PR abaixo. Acredito que será aceito pelo projeto.
https://github.com/wmixvideo/nfe/pull/839

**Reply by edivaldoramos** [reply]
Vou fazer isso @aguinaldofryder.
Valeu !
